libvcx/src/proof.rs

```diff
@@ -864,6 +864,28 @@ pub mod tests {
         assert_eq!(proof.state(), VcxStateType::VcxStateOfferSent as u32);
     }
 
+    #[test]
+    #[cfg(feature = "general_test")]
+    fn test_proof_update_state_v2() {
+        let _setup = SetupAriesMocks::init();
+
+        let connection_handle = build_test_connection();
+
+        let mut proof = Verifier::create("1".to_string(),
+                                  REQUESTED_ATTRS.to_owned(),
+                                  REQUESTED_PREDICATES.to_owned(),
+                                  r#"{"support_revocation":false}"#.to_string(),
+                                  "Optional".to_owned()).unwrap();
+
+        proof.send_presentation_request(connection_handle).unwrap();
+
+        assert_eq!(proof.state(), VcxStateType::VcxStateOfferSent as u32);
+
+        proof.update_state(Some(PROOF_RESPONSE_DECRYPTED), Some(connection_handle)).unwrap();
```

Comment:
missing `deserialization`, `release`, `serialization` before updating the state here